lib/wanda/execution/evaluation.ex

```diff
@@ -71,6 +74,16 @@ defmodule Wanda.Execution.Evaluation do
     }
   end
 
+  defp add_facts(%AgentCheckResult{} = agent_check_result, facts, check_id) do
```

Comment:
Why do you need to receive the `check_id` as an independent parameter?
The `check_id` already comes in the facts result together with the name and value

Reply:
The facts variable here is a hashmap that we use internally in gathering business logic as a way to organize data to be accessible in an easy way.
It does not have any reference to the `check_id` field. In this function, we rebuild the original struct to send it over alongside the results for debugging/ux purposes.